### ISSUE TITLE
[BUG]: MOE Failed with PyTorch 1.8

### ISSUE THREAD

**Issue by FrankLeeeee** [hpcaitech/ColossalAI#736]
### 🐛 Describe the bug

When running unit test with torch 1.8, the unit tests for moe module failed as shown below. 

<img width="637" alt="Screenshot 2022-04-12 at 4 21 19 PM" src="https://user-images.githubusercontent.com/31818963/162915139-7423b24f-0f0c-4378-9063-c56c290a3941.png">

The error occurs because the API of `torch.nn.Linear` is different between PyTorch 1.8 and latest PyTorch. 

```python
import torch
import torch.nn as nn

# allowed in torch 1.8
layer = nn.Linear(128, 512)

# NOT allowed in torch 1.8
layer = nn.Linear(128, 512, device=torch.cuda.current_device())


# allowed in torch 1.9 onwards
layer = nn.Linear(128, 512)

# allowed in torch 1.9 onwards
layer = nn.Linear(128, 512, device=torch.cuda.current_device())
```

There are two cases this might lead to an exception.

1. directly pass `device` to `torch.nn.Linear`
2. initialize the`torch.nn.Linear` under the context of `with seed(ParallelMode.TENSOR)`. This context is used to allow parameters to initialize differently across parallel mode and this is currently only supported on layers which can be initialized on GPUs.

### Environment

PyTorch: 1.8.1
CUDA: 11.1.1

**Comment by binmakeswell** [comment]
https://github.com/hpcaitech/ColossalAI#Installation
We have updated a lot. This issue was closed due to inactivity. Thanks.
